### ISSUE TITLE
Support for Python

### ISSUE THREAD

**Issue by billandr** [davidalger/warden#368]
I'm thinking about adding support for a python, oracle, redis, nginx environment.  Has this already been done / considered?  Most of my work will seem to be needed within the env command.

Otherwise; I would need to have a docker-composer.yaml plug into warden traffik for dns resolution and other containers like nginx etc. (e.g. building the docker compose to support the warden nginx and warden redis but my own python and oracle images) - which is also possible.  Not sure what others have done about other custom containers.

**Comment by navarr** [comment]
You could start with a `local` environment type (which excludes all services), and then add additional services as specified in [Customizing an Environment](https://docs.warden.dev/environments/customizing.html).

.env
```
WARDEN_NGINX=1
WARDEN_REDIS=1
```

Then you would create a `.warden/warden-env.yml` file that would be merged with all the other docker files. That could quickly get you setup with Nginx and Redis at least, and then you'd only need to add Python and Oracle

**Comment by billandr** [comment]
yes; that's exactly how I started; and then realized that there was a dependency on php-fpm container (name); technically if I use that name for my python app it all works.  So that so far was my only "tweak" to make work.  To describe; I was pulling in nginx, db (but using custom image), new service for python, redis and it wouldn't bring up due to missing php-fpm; but if I rename python to php-fpm it all works.  I think it was nginx that required php-fpm.  

Like you did with mutagen; I was thinking that maybe adding an environment override as well that if an environment type existed in the folder that it would use it in order to customize some of this; but maybe not necessary.  What's the easiest way to remove the php-fpm dependency from the nginx without build a new one?

**Comment by billandr** [comment]
To add additional information; so I'm able to get everything to work; with the exception that php-fpm is hardcoded for the mutagen sync commands.  So do you know of an easy way to get the "app" container (which would be php-fpm or python-app or whatever someone decides to call it)?  If so; then that could be used to dynamically set the name so that it's not looking for exactly the text php-fpm but instead whatever was labeled to be used.

For now; I'll keep it as php-fpm since everything works as expected with the exception of the "name"; and changing those two files isn't worth the cosmetic.  Maybe being more generic instead of php-fpm but app and app-debug or web and web-debug for the two container types?

**Comment by navarr** [comment]
@billandr: I'm having trouble identifying where php-fpm is hardcoded.  Could you please point it out in the repository?

Are you referring to the automatic `warden sync start/resume` parts of the env command?

If you're referring to the [default.conf](https://github.com/davidalger/warden/blob/684d248b3cfa6eecc56542db450319a6e803af6e/images/nginx/context/etc/nginx/conf.d/default.conf) you could mount an overriding file, which I can understand isn't as nice but is a decent workaround

**Comment by billandr** [comment]
you are correct; it's in the env.cmd (with the mutagen / sync commands); and only once in the "sync.cmd".  I used a variable at the top of the file and referenced that variable in all the places and that worked for me.  So technically - being able to query for some name from some setting should solve for using any name.
